src/Domain/Donation.php

```diff
@@ -442,35 +442,11 @@ public function replaceNullPaymentMethodTypeWithCard(): void
         $this->paymentMethodType = PaymentMethodType::Card;
     }
 
-    /**
-     * @return array A json encode-ready array representation of the donation, for sending to Salesforce.
-     */
-    public function toHookModel(): array
-    {
-        $data = $this->toApiModel();
-
-        $data['updatedTime'] = $this->getUpdatedDate()->format(DateTimeInterface::ATOM);
-        $data['amountMatchedByChampionFunds'] = (float) $this->getConfirmedChampionWithdrawalTotal();
-        $data['amountMatchedByPledges'] = (float) $this->getConfirmedPledgeWithdrawalTotal();
-        $data['originalPspFee'] = (float) $this->getOriginalPspFee();
-        $data['refundedTime'] = $this->refundedAt?->format(DateTimeInterface::ATOM);
-        $data['tbgGiftAidRequestConfirmedCompleteAt'] =
-            $this->tbgGiftAidRequestConfirmedCompleteAt?->format(DateTimeInterface::ATOM);
-        unset(
-            $data['charityName'],
-            $data['donationId'],
-            $data['matchReservedAmount'],
-            $data['matchedAmount'],
-            $data['cardBrand'],
-            $data['cardCountry'],
-        );
-
-        return $data;
-    }
-
-    public function toApiModel(): array
+    public function toApiModel(bool $forSalesforce = false): array
```

Comment:
I'm not keen on the flag argument, I think it'd be clearer to have two functions, e.g. `toFEApiModel` and `toSFApiModel`. The latter is easy to implement on top of the former.  Easier to see `toSFApiModel` is used that way.

Reply:
If you really want a flag arg then maybe make it an enum instead of a bool, and at least then it'd be easy to find usages of the `ApiConsumer::salesforce` case.

Reply:
But I think the non-flag way was more or less what we had before, so would be good to understand the reasoning for the change.